src/protocol/boolean/bit_decomposition.rs

```diff
@@ -21,17 +22,15 @@ pub struct BitDecomposition {}
 impl BitDecomposition {
     #[allow(dead_code)]
     pub async fn execute<F: Field>(
+        rbg: RandomBitsGenerator<F>,
         ctx: SemiHonestContext<'_, F>,
         record_id: RecordId,
         a_p: &Replicated<F>,
```

Comment:
nit: Other protocols take `ctx` and `record_id` as first two arguments, followed by inputs. Maybe change the order to:
```
ctx: SemiHonestContext<'_, F>,
record_id: RecordId,
rbg: RandomBitsGenerator<F>,
a_p: &Replicated<F>,
```